main/src/main/scala/sbt/plugins/SemanticDbPlugin.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * sbt
+ * Copyright 2011 - 2018, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * Licensed under Apache License 2.0 (see LICENSE)
+ */
+
+package sbt
+package plugins
+
+import Keys._
+import sbt.librarymanagement.syntax._
+import sbt.librarymanagement.CrossVersion
+import Project.inConfig
+import sbt.io.IO
+
+object SemanticDBPlugin extends AutoPlugin {
+  override def requires = JvmPlugin
+  override def trigger = allRequirements
+
+  override lazy val globalSettings = Seq(
+    enableSemanticDB := false,
+    semanticDBIncludeInJar := false,
+    semanticDBOptions := List("-Yrangepos"),
+    semanticDBVersion := "4.0.0",
+    semanticDBCompilerPlugin := {
+      val v = semanticDBVersion.value
+      ("org.scalameta" % "semanticdb-scalac" % v).cross(CrossVersion.full)
+    }
+  )
+
+  override lazy val projectSettings = Seq(
+    libraryDependencies ++= {
```

Comment:
Does this work if users do `libraryDependencies :=` or `scalacOptions :=`? My experience with the previous sbt-scalafix version is that quite many builds do that and when it happens the user has no clue how to even debug the problem. The current sbt-scalafix installation instructions use `addCompilerPlugin` and `scalacOptions +=` and my impression so far is that people at least understand how that works so they can debug it.

Reply:
Good point. I can try `allDependencies`.